portal/get_patient_documents.php

```diff
@@ -1,160 +1,131 @@
-                $zip->addFromString(str_replace($source . '/', '', $file), file_get_contents($file));
+        // Function to validate form submission
+        function validateForm(event) {
+            let checkboxes = document.querySelectorAll('input[type="checkbox"]:checked');
+            if (checkboxes.length === 0) {
+                alert("Please select at least one document to download.");
+                event.preventDefault(); // Prevent form submission
             }
         }
-    } elseif (is_file($source) === true) {
-        $zip->addFromString(basename($source), file_get_contents($source));
-    }
-
-    return $zip->close();
-}
+    </script>
+</head>
+<body>
+    <div class="container-fluid">
+        <h4><?php echo text("Select Documents to Download"); ?></h4>
+        <form id="download-form" action="./report/document_downloads_action.php" method="post" onsubmit="validateForm(event)">
+            <input type="hidden" name="csrf_token_form" value="<?php echo attr(CsrfUtils::collectCsrfToken()); ?>" />
+            <div class="form-check mb-3">
+                <input class="form-check-input" type="checkbox" id="selectAll" onclick="toggleAllCheckboxes(this)">
+                <label class="form-check-label" for="selectAll"><?php echo text("Select All Toggle"); ?></label>
+            </div>
+            <div class="row">
+                <?php foreach ($documents as $category => $docs) {
+                    $title = str_replace("Categories/", "", $category);
+                    $normalizedTitle = preg_replace('/[^a-zA-Z0-9]/', '-', $title);
+                    ?>
+                    <div class="col-md-6">
+                        <div class="card mb-4">
+                            <div class="card-header d-flex justify-content-between align-items-center">
+                                <h5 class="mb-0"><?php echo text($normalizedTitle); ?></h5>
+                                <div class="form-check">
+                                    <input class="form-check-input" type="checkbox" id="selectAll<?php echo attr($normalizedTitle); ?>" onclick="toggleCheckboxes('category-<?php echo attr($normalizedTitle); ?>', this)">
+                                    <label class="form-check-label" for="selectAll<?php echo attr($normalizedTitle); ?>"><?php echo text("Select All"); ?></label>
+                                </div>
+                            </div>
+                            <div class="card-body">
+                                <?php foreach ($docs as $doc) { ?>
+                                    <div class="form-check">
+                                        <input class="form-check-input category-<?php echo attr($normalizedTitle); ?>" type="checkbox" name="documents[]" value="<?php echo attr($doc['id']); ?>" id="doc<?php echo text($doc['id']); ?>">
+                                        <label class="form-check-label" for="doc<?php echo attr($doc['id']); ?>">
```

Comment:
id should be attr instead of text

Reply:
is a text label

Reply:
to clarify, i meant the `id="doc<?php echo text($doc['id']); ?>">`
this is an html attribute, so needs attr()